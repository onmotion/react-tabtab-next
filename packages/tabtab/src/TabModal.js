"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var react_poppop_1 = __importDefault(require("react-poppop"));
var react_sortable_hoc_1 = require("react-sortable-hoc");
var SortMethod_1 = __importDefault(require("./SortMethod"));
var DragTabContainer = (0, react_sortable_hoc_1.SortableContainer)(function (_a) {
    var children = _a.children;
    return (React.createElement("div", { style: { marginTop: '50px' } }, children));
});
var ModalTabListWrapper = (function (_super) {
    __extends(ModalTabListWrapper, _super);
    function ModalTabListWrapper() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    ModalTabListWrapper.prototype.render = function () {
        return (React.createElement(DragTabContainer, { onSortEnd: this.onSortEnd, axis: 'y', lockAxis: 'y', pressDelay: 100 }, this.props.children));
    };
    return ModalTabListWrapper;
}(SortMethod_1.default));
var TabModal = (function (_super) {
    __extends(TabModal, _super);
    function TabModal() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    TabModal.prototype.render = function () {
        return (React.createElement(react_poppop_1.default, { open: true, onClose: this.props.closeModal, closeOnEsc: true, closeBtn: true },
            React.createElement(ModalTabListWrapper, { handleTabSequence: this.props.handleTabSequence, handleTabChange: this.props.handleTabChange, activeIndex: this.props.activeIndex }, this.props.children)));
    };
    return TabModal;
}(React.Component));
exports.default = TabModal;
