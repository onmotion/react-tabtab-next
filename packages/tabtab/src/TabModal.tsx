import * as React from 'react';

import SortMethod from './SortMethod';
import ReactModal from 'react-modal';
import styled from 'styled-components';

ReactModal.setAppElement('#root');

type Props = {
    closeModal: (event: any) => void;
    handleTabSequence: (event: any) => void;
    handleTabChange: (event: any) => void;
    activeIndex: number;
    children: React.ReactNode[];
};

class ModalTabListWrapper extends SortMethod {
    render() {
        return <></>;
    }
}

const Modal = styled(ReactModal)`
    margin: auto;
    background-color: #fff;
    width: 200px;
    padding: 20px;
    margin-top: 40px;
    border: 1px solid #00000022;
`;
export default class TabModal extends React.Component<Props> {
    render() {
        return (
            <Modal isOpen={true} contentLabel="Modal" onRequestClose={this.props.closeModal} className="tabtab-modal">
                <ModalTabListWrapper
                    handleTabSequence={this.props.handleTabSequence}
                    handleTabChange={this.props.handleTabChange}
                    activeIndex={this.props.activeIndex}
                >
                    {this.props.children}
                </ModalTabListWrapper>
            </Modal>
        );
    }
}
