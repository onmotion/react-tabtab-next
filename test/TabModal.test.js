import React from 'react';
import TabModal from '../src/TabModal';
import {shallow} from 'enzyme';
import noop from 'noop3';

test('render Panel', () => {
  const component = shallow(
    <TabModal closeModal={noop}
              handleTabSequence={noop}
              handleTabChange={noop}>
      <span>fake child</span>
    </TabModal>
  );
  expect(component).toMatchSnapshot();
})
