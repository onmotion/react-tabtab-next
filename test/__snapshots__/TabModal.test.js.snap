// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Panel 1`] = `
<PopPop
  closeBtn={true}
  closeOnEsc={true}
  closeOnOverlay={true}
  onClose={[Function]}
  open={true}
  overlayStyle={Object {}}
  position="topCenter"
>
  <ModalTabListWrapper
    handleTabChange={[Function]}
    handleTabSequence={[Function]}
  >
    <span>
      fake child
    </span>
  </ModalTabListWrapper>
</PopPop>
`;
